yeastphenome/apps/api/datasets.py

```diff
@@ -404,33 +401,32 @@ class GetSimilarGenes(RatelimitMixin, APIView):
     ratelimit_method = "GET"
     renderer_classes = (JSONRenderer,)
 
-    def get(self, request, systematic_name, N=10, reverse=0):
+    def get(self, request, gene_id, N=10, reverse=0):
         print("GET GetSimilarGenes")
         try:
-            gene = Gene.objects.get(systematic_name=systematic_name)
+            gene = Gene.objects.get(pk=gene_id)
         except Gene.DoesNotExist:
             return Response(status=404)
 
-        # Only get top and bottom N
-        sims = list(gene.get_ranked_similar(reverse=(reverse == 1)))
-        first_n = sims[:N]
-        last_n = sims[len(sims) - N :]
-
-        scores = {}
-        for sim in first_n + last_n:
-            if sim.gene1.systematic_name == gene.systematic_name:
-                scores[sim.gene2.systematic_name] = {
-                    "score": sim.score,
-                    "pvalue": sim.pvalue,
-                }
-            else:
-                scores[sim.gene1.systematic_name] = {
-                    "score": sim.score,
-                    "pvalue": sim.pvalue,
-                }
+        # # Only get top and bottom N
```

Comment:
Did you mean to uncomment this? It looks like you generate the scores in a more Pythonic way, and if you've tested this feel free to remove the old (now commented) version.

Reply:
Yep, I forgot to delete these lines, after commenting them out for testing. I'll push an update to the PR after I've made adjustments in response to your comments.